rs/crypto/tests/vetkd.rs

```diff
@@ -7,98 +10,680 @@ use ic_crypto_test_utils_ni_dkg::{
-            .expect("failed to deserialize encrypted VetKey");
-
-        let derived_public_key = ic_vetkeys::DerivedPublicKey::deserialize(&derived_public_key)
-            .expect("failed to deserialize derived public key");
-        let decrypted_key = encrypted_key
-            .decrypt_and_verify(
-                &transport_secret_key,
-                &derived_public_key,
-                &vetkd_args.input,
-            )
-            .expect("failed to decrypt vetKey")
+        Ok(encrypted_key
+            .decrypt_and_verify(&self.tsk, &self.dk, &self.input)?
             .signature_bytes()
-            .to_vec();
+            .to_vec())
+    }
+}
+
+fn wrong_ni_dkg_id(dkg_id: &NiDkgId) -> NiDkgId {
+    NiDkgId {
+        dealer_subnet: dkg_id.dealer_subnet,
+        dkg_tag: dkg_id.dkg_tag.clone(),
+        target_subnet: dkg_id.target_subnet,
+        start_block_height: (dkg_id.start_block_height.get() + 1000000).into(),
+    }
+}
+
+#[test]
+fn should_vetkd_consistently_derive_the_same_vetkey_given_sufficient_shares() {
+    let mut rng = reproducible_rng();
+    let server = VetKDTestServer::new(&mut rng);
+    let client = VetKDTestClient::new(&mut rng, &server);
+    let vetkd_args = client.create_args(&server.dkg_id);
+
+    let mut keys = vec![];
+
+    for _i in 0..3 {
+        let shares = server
+            .create_key_shares(&vetkd_args, &mut rng)
+            .expect("Share creation failed");
+
+        assert!(
+            server
+                .verify_key_shares(&shares, &vetkd_args, &mut rng)
+                .is_ok()
+        );
+
+        let encrypted_key = server
+            .combine_key_shares(&shares, &vetkd_args, &mut rng)
+            .expect("Share combination failed");
+
+        let decrypted_key = client
+            .decrypt_key(&encrypted_key)
+            .expect("Failed to decrypt key");
+
+        keys.push(decrypted_key);
+    }
+
+    assert!(keys.iter().all(|k| *k == keys[0]));
+}
+
+#[test]
+fn should_vetkd_create_encrypted_key_share_err_if_threshold_sig_data_not_loaded() {
+    let mut rng = reproducible_rng();
+    let server = VetKDTestServer::new(&mut rng);
+    let client = VetKDTestClient::new(&mut rng, &server);
+    let vetkd_args = client.create_args(&wrong_ni_dkg_id(&server.dkg_id));
+
+    let shares = server.create_key_shares(&vetkd_args, &mut rng);
+
+    match shares {
+        Err(VetKdKeyShareCreationError::ThresholdSigDataNotFound(_)) => { /* expected */ }
+        Ok(_) => panic!("Unexpected success"),
+        Err(e) => panic!("Unexpected error {:?}", e),
+    }
+}
+
+#[test]
+fn should_vetkd_create_encrypted_key_share_err_if_pub_coeffs_in_store_are_empty() {
+    let mut rng = reproducible_rng();
+    let mut server = VetKDTestServer::new(&mut rng);
+    let client = VetKDTestClient::new(&mut rng, &server);
+    let vetkd_args = client.create_args(&server.dkg_id);
+
+    server.modify_stored_keys(&mut rng, |_sks, _pks| {
+        todo!("how to modify store?");
```

Comment:
It is my understanding that this and all other uncommented `todo!`'s are an instance of [this](https://github.com/dfinity/ic/pull/7098/files#r2414249727).